src/codegen/sys/lib_.rs

```diff
@@ -444,7 +417,7 @@ fn generate_records(w: &mut Write, env: &Env, records: &[&Record]) -> Result<()>
         if lines.is_empty() {
             try!(writeln!(
                 w,
-                "{}#[repr(C)]\n{0}pub struct {}(c_void);\n",
+                "{0}#[repr(C)]\n{0}#[derive(Copy,Clone)]\n{0}pub struct {1}(u8);\n",
```

Comment:
Ah, so this is just for having *anything* in case of opaque structs. Maybe make those `pub struct Meh(())`? That way they can only used as pointers (which they can only be used as correctly anyway).